### PR TITLE
refactor(angular): tag merge

### LINKED ISSUE #518: Feature(business): Tag
https://digital.sbb.ch/de/webapps/components/tag

### PR DESCRIPTION
**Pull request by jeripeierSBB**
Closes #518 , #485

### CODE REVIEW

File: src/angular/tag/tag-link.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  /** Amount displayed in badge */
+  @Input()
+  get amount(): number {
+    return this._amount;
+  }
+  set amount(value: number) {
+    this._amount = coerceNumberProperty(value);
+    this._badgeDescriptionFallback =
+      typeof $localize === 'function'
+        ? $localize`:Aria label for amount of results displayed in badge pill@@sbbTagBadgePillAmountOfResults:${this.amount} results available`
```
Comment (jeripeierSBB) [inline src/angular/tag/tag-link.ts:28]: @kyubisation  Does this need a different i18n ID than sbb-tag ?

Comment (kyubisation) [reply src/angular/tag/tag-link.ts:28]: If it is exactly the same text and has the same meaning, no. Otherwise, yes.

Comment (jeripeierSBB) [reply src/angular/tag/tag-link.ts:28]: it's exactly the same

File: src/angular/tag/tag.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  _setCheckedAndEmit(checked: boolean) {
+    const previousChecked = this.checked;
+    this.checked = checked;
+    if (previousChecked !== this.checked) {
+      this._emitChangeEvent();
+    }
+  }
+
+  ngOnDestroy() {
+    this._amountChange.complete();
```
Comment (kyubisation) [inline src/angular/tag/tag.ts:124]: Missing super.ngOnDestroy() call

File: src/components-examples/angular/tag/tag-template-forms/tag-template-forms-example.html [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<sbb-tags>
+  <sbb-tag [(ngModel)]="trains" label="Trains" amount="23"></sbb-tag>
+  <sbb-tag [(ngModel)]="cars" label="Cars" amount="2"></sbb-tag>
+  <sbb-tag [(ngModel)]="bicycles" label="Bicycles" amount="150"></sbb-tag>
```
Comment (goestu) [inline src/components-examples/angular/tag/tag-template-forms/tag-template-forms-example.html:4]:
```suggestion
  <sbb-tag [(ngModel)]="trains" amount="23">Trains</sbb-tag>
  <sbb-tag [(ngModel)]="cars" amount="2">Cars</sbb-tag>
  <sbb-tag [(ngModel)]="bicycles" amount="150">Bicycles</sbb-tag>
```

**Review by kyubisation — COMMENTED** [review]
As discussed, replace label with ng-content

File: src/showcase/app/public/public-examples/tag-examples/tag-advanced-example/tag-advanced-example.component.html [hunk 4]
```diff
@@ -0,0 +1,25 @@
+<sbb-tags>
+  <sbb-tag
+    *ngFor="let tag of tags; let i = index"
+    (change)="change($event)"
+    [label]="tag.label"
+    [amount]="tag.amount"
+  ></sbb-tag>
```
Comment (goestu) [inline src/showcase/app/public/public-examples/tag-examples/tag-advanced-example/tag-advanced-example.component.html:7]:
```suggestion
    [amount]="tag.amount"
  >tag.label</sbb-tag>
```

File: src/showcase/app/public/public-examples/tag-examples/tag-examples.module.ts [hunk 5]
```diff
@@ -8,12 +8,23 @@ import { SbbTagModule } from '@sbb-esta/angular-public/tag';
 
 import { provideExamples } from '../../../shared/example-provider';
 
-import { TagExampleComponent } from './tag-example/tag-example.component';
+import { TagAdvancedExample } from './tag-advanced-example/tag-advanced-example.component';
+import { TagLinkExample } from './tag-link-example/tag-link-example.component';
+import { TagReactiveFormsExample } from './tag-reactive-forms-example/tag-reactive-forms-example.component';
+import { TagTemplateFormsExampleComponent } from './tag-template-forms-example/tag-template-forms-example.component';
```
Comment (goestu) [inline src/showcase/app/public/public-examples/tag-examples/tag-examples.module.ts:14]: In https://github.com/sbb-design-systems/sbb-angular/blob/master/MERGE_MIGRATION.md at step 1:
Rename files with *.component.* pattern to *.* and update references (e.g. autocomplete.component.ts should become autocomplete.ts
Are examples excludet from this rule?

Comment (jeripeierSBB) [reply src/showcase/app/public/public-examples/tag-examples/tag-examples.module.ts:14]: these are the old examples which can be left as they are, but I see that they don't have consistent Component suffix. I changed it :-)

File: src/angular/tag/tag.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+#### Total Amount in All-Tag
+
+By default, the amount badge in the 'all'-tag is calculated by summing up all amounts of every child tag component.
+If you like to set the total amount manually (e.g. if the amounts in the tag components are not exclusively)
+then you have to set the `totalAmount` property of `<sbb-tags>`.
+
+```html
+<sbb-tags [totalAmount]="100">
+  <sbb-tag label="Trains" [amount]="10" [(ngModel)]="tag"></sbb-tag>
```
[the remark this answers is not in the document]
Comment (jeripeierSBB) [reply src/angular/tag/tag.md:28]: This change was only for the link tag, so it's correct as it is. But a few minutes ago @kyubisation and I decided to change that for sbb-tag usages too. So soon, you'll see the changes ;)

**Review by kyubisation — APPROVED** [review]
LGTM 👍 
Thank you both @jeripeierSBB and @goestu 😃

**Pull request merged**